test/integration_test/migration_dr_actions_test.go

```diff
@@ -0,0 +1,517 @@
+
+	t.Run("testDRActionFailoverMultipleNamespacesTest", testDRActionFailoverMultipleNamespacesTest)
+	t.Run("testDRActionFailoverSubsetNamespacesTest", testDRActionFailoverSubsetNamespacesTest)
+	t.Run("testDRActionFailoverWithMigrationRunningTest", testDRActionFailoverWithMigrationRunningTest)
+
+}
+
+// testDRActionFailoverMultipleNamespacesTest tests failover action for multiple namespaces
+func testDRActionFailoverMultipleNamespacesTest(t *testing.T) {
+	var testrailID, testResult = 297512, testResultFail
+	runID := testrailSetupForTest(testrailID, &testResult, t.Name())
+	defer updateTestRail(&testResult, testrailID, runID)
+	defer updateDashStats(t.Name(), &testResult)
+
+	// Create the apps for migration
+	// Start migrationschedule
+	instanceID := "multi-ns-migration-schedule-interval"
+	// appKeys is an array of appKeys that will be used to create the apps
+	mysqlApp := "mysql-1-pvc"
+	mysqlNamespace := fmt.Sprintf("%s-%s", mysqlApp, instanceID)
+	elasticsearchApp := "elasticsearch"
+	elasticsearchNamespace := fmt.Sprintf("%s-%s", elasticsearchApp, instanceID)
+	appKeys := []string{mysqlApp, elasticsearchApp}
+	var err error
+	// Reset config in case of error
+	defer func() {
+		err = setSourceKubeConfig()
+		log.FailOnError(t, err, "Error resetting remote config")
+	}()
+
+	// Schedule multiple apps
+	var preMigrationCtxs []*scheduler.Context
+	for _, appKey := range appKeys {
+		ctxs, err := schedulerDriver.Schedule(instanceID,
+			scheduler.ScheduleOptions{
+				AppKeys: []string{appKey},
+				Labels:  nil,
+			})
+		log.FailOnError(t, err, "Error scheduling task")
+		Dash.VerifyFatal(t, 1, len(ctxs), "Only one task should have started")
+		preMigrationCtxs = append(preMigrationCtxs, ctxs[0].DeepCopy())
+	}
+
+	for _, preMigrationCtx := range preMigrationCtxs {
+		err = schedulerDriver.WaitForRunning(preMigrationCtx, defaultWaitTimeout, defaultWaitInterval)
+		log.FailOnError(t, err, "Error waiting for app to get to running state")
+	}
+
+	sourceDeployments, err := apps.Instance().ListDeployments(mysqlNamespace, metav1.ListOptions{})
+	log.FailOnError(t, err, "error retrieving deployments from %s namespace", mysqlNamespace)
+	Dash.VerifyFatal(t, len(sourceDeployments.Items), 1, fmt.Sprintf("Expected 1 deployment in source in %s namespace", mysqlNamespace))
+	sourceDeploymentReplicas := *sourceDeployments.Items[0].Spec.Replicas
+	sourceStatefulsets, err := apps.Instance().ListStatefulSets(elasticsearchNamespace, metav1.ListOptions{})
+	log.FailOnError(t, err, "error retrieving statefulsets from %s namespace", elasticsearchNamespace)
+	Dash.VerifyFatal(t, len(sourceStatefulsets.Items), 1, fmt.Sprintf("Expected 1 statefulset in source in %s namespace", elasticsearchNamespace))
+	sourceStatefulsetReplicas := *sourceStatefulsets.Items[0].Spec.Replicas
+
+	// Create the clusterpair
+	clusterPairNamespace := defaultAdminNamespace
+	log.Info("Bidirectional flag is set, will create bidirectional cluster pair:")
```

Comment:
Just to clarify, for these tests the new jenkins job created will be hardcoded with bidirectional flag right?
Then for the unidirectional cases we will have to create a separate test suite and job

Reply:
Yes, priority is bidirectional. For now the jobs will not have any option to choose. In future, we may consider adding unidirectional, if it's really required. For failback, also our recommendation is to always use bidirectional pairing.